### DISCUSSION TITLE
Is it possible to group Dockerfile and docker-compose.yml updates?

### DISCUSSION THREAD

**Discussion by Masa-Shin** [renovatebot/renovate#8086] [Request Help]
I've tried to group all docker related updates to one PR but renovate bot created separated PRs for them. Here is my renovate.json.

```
{
  "extends": [
    "config:base"
  ],

  "enabledManagers": ["npm", "dockerfile", "docker-compose"],

  "suppressNotifications": ["prIgnoreNotification"],
  "prCreation": "immediate",
  "unpublishSafe": true,
  "draftPR": true,
  "labels": ["dependencies"],
  "vulnerabilityAlerts": {
    "labels": ["security"]
  },

  "schedule": [
    "at any time"
  ],
  "timezone": "Asia/Tokyo",

  "node": {
    "supportPolicy": ["lts_latest"]
  },

  "packageRules": [
    {
      "paths": ["+(package.json)"],
      "postUpdateOptions": ["yarnDedupeFewer"]
    },
    {
      "paths": ["+(package.json)"],
      "updateTypes": ["major"],
      "groupName": "npm major updates",
      "groupSlug": "npm-major-updates"
    },
    {
      "paths": ["+(package.json)"],
      "updateTypes": ["patch", "minor"],
      "groupName": "patch & minor npm updates",
      "groupSlug": "npm-patch-minor-updates",
      "prPriority": -1
    },

    {
      "packagePatterns": ["docker.*"],
      "updateTypes": ["major"],
      "groupName": "docker related major updates",
      "groupSlug": "docker-related-major-updates"
    },
    {
      "packagePatterns": ["docker.*"],
      "updateTypes": ["patch", "minor"],
      "groupName": "docker related patch & minor updates",
      "groupSlug": "docker-related-patch-minor-updates",
      "prPriority": -1
    }
  ]
}
```

The bot created [Update node Docker tag to v14.15.3] PR for Dockerfile and [Update redis Docker tag to v5.0.10] for docker-compose.yml, both of them are patch updates.

Is it possible to have just one PR for those kind of files if all updates are patch or minor?

Thanks in advance.

**Comment by rarkins** [comment]
Using `packagePatterns` is wrong. That is meant to refer to the package name, which is e.g. `redis`. Replace it with `"datasources": ["docker"]`

**Reply by Masa-Shin** [reply]
Thank you so much! I will try this.

**Reply by Masa-Shin** [reply]
@rarkins 
Unfortunately, changing to `"datasources": ["docker"]` doesn't work. [Here is a replication repo](https://github.com/Masa-Shin/word_mastermind3/pull/1) for the issue. It says it will create two PRs related to docker: 
- [Update node Docker tag to v14.15.3] for Dockerfile
- [Update redis Docker tag to v5.0.10] for docker-compose.yml

**Reply by viceice** [reply]
I think you need to override additional branch prefix, as it is set by docker file manager.

**Reply by Masa-Shin** [reply]
@viceice 
Thanks! I added it but it seems additional prefix only applies for the second PR.
<img width="417" alt="スクリーンショット 2021-01-03 0 30 37" src="https://user-images.githubusercontent.com/34234442/103460607-ed62c280-4d5a-11eb-825a-429a24be49d1.png">

**Reply by Masa-Shin** [reply]
I don't know why changing to `"datasources"` does not resolve the problem. Should I make an issue for this?

**Reply by viceice** [reply]
You need to set the branch prefix for docker file and docker compose manager to apply to both. The alternative option is to apply the prefix to the docker datasource, so it should be applied to all docker image deps.

**Reply by viceice** [reply]
@Masa-Shin can you provide a minimal test repo with only the docker, compose and renovate files, so we can run some debugging against?

**Reply by Masa-Shin** [reply]
@viceice 
Sure. [Here](https://github.com/Masa-Shin/repro_renovate_disccussion_8086) it is.

**Reply by viceice** [reply]
Thanks, will try to run a debug session later
